clients/src/test/java/org/apache/kafka/test/MockConsumerInterceptor.java

```diff
@@ -52,6 +59,9 @@ public void configure(Map<String, ?> configs) {
 
     @Override
     public ConsumerRecords<String, String> onConsume(ConsumerRecords<String, String> records) {
+        if (CLUSTER_META.get() != null && CLUSTER_META.get().clusterId() != null && CLUSTER_META.get().clusterId().length() == 48)
+            IS_CLUSTER_ID_PRESENT_BEFORE_ON_CONSUME.set(true);
```

Comment:
Instead of doing this, you can just set the cluster id on another variable called `CLUSTER_META_BEFORE_ON_CONSUME` via a compare and set (so that it's only done once).